File: src/test/java/io/vproxy/vfx/intro/DemoVScene.java
package io.vproxy.vfx.intro;

import io.vproxy.vfx.ui.scene.VScene;
import io.vproxy.vfx.ui.scene.VSceneRole;

public abstract class DemoVScene extends VScene {
    public DemoVScene(VSceneRole role) {
        super(role);
    }

    public abstract String title();
}

File: src-ts/src/intro/DemoVScene.ts
import { VScene } from '../ui/scene/VScene.js';
import { VSceneRole } from '../ui/scene/VSceneRole.js';

export abstract class DemoVScene extends VScene {
  constructor(role: VSceneRole) {
    super(role);
  }

  abstract title(): string;
}
